File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/domain/OptionDetail.java
package io.beaniejoy.dongnecafe.domain.cafe.domain;

import io.beaniejoy.dongnecafe.domain.cafe.dto.menu.OptionDetailResponseDto;
import io.beaniejoy.dongnecafe.common.domain.BaseTimeEntity;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class OptionDetail extends BaseTimeEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long optionDetailId;

    private String name;

    private Integer extra;

    @ManyToOne
    @JoinColumn(name = "option_id")
    private MenuOption menuOption;

    public OptionDetailResponseDto toResponseDto() {
        return OptionDetailResponseDto.builder()
                .optionDetailId(optionDetailId)
                .name(name)
                .extra(extra)
                .build();
    }
}

File: src/main/java/io/beaniejoy/dongnecafe/domain/cafe/entity/OptionDetail.kt
package io.beaniejoy.dongnecafe.domain.cafe.entity

import io.beaniejoy.dongnecafe.common.entity.BaseTimeEntity
import java.math.BigDecimal
import javax.persistence.*

@Entity
@Table(name = "option_detail")
class OptionDetail protected constructor(
    name: String,
    extraPrice: BigDecimal
) : BaseTimeEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "option_detail_id", nullable = false)
    val id: Long = 0L

    @Column(name = "name", nullable = false)
    val name: String = name

    @Column(name = "extra_price", nullable = false)
    val extraPrice: BigDecimal = extraPrice

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "menu_option_id", nullable = false)
    var menuOption: MenuOption? = null
        protected set

    companion object {
        fun createOptionDetail(name: String, extraPrice: BigDecimal): OptionDetail {
            return OptionDetail(
                name = name,
                extraPrice = extraPrice
            )
        }
    }

    fun updateMenuOption(menuOption: MenuOption) {
        this.menuOption = menuOption
    }
}